scripts/globals/znm.lua

```diff
@@ -16,27 +17,98 @@ require("scripts/globals/msg")
 require("scripts/globals/npc_util")
 require("scripts/globals/pankration")
 require("scripts/globals/utils")
+require("scripts/globals/znm_data")
 -----------------------------------
 
 xi = xi or {}
 xi.znm = xi.znm or {}
 
+--------------------------------------------------
+-- Sanraku's Interest and Recommended Fauna
+-- Applies bonuses to soul plate zeni-value
+--------------------------------------------------
+-- Called during JstMidnight tick
+xi.znm.UpdateSanrakusMobs = function()
+    SetServerVariable('[ZNM][Sanraku]Interest', math.random(#xi.znm.SANRAKUS_INTEREST))
+    SetServerVariable('[ZNM][Sanraku]Fauna', math.random(#xi.znm.SANRAKUS_FAUNA))
+end
+
+-- Get Sanraku's "Subject of Interest"
+xi.znm.getSanrakusInterest = function()
+    local interest = GetServerVariable('[ZNM][Sanraku]Interest')
+
+    -- Initialize the server var if it hasn't been already
+    if interest == nil or interest == 0 then
+        interest = math.random(#xi.znm.SANRAKUS_INTEREST)
+        SetServerVariable('[ZNM][Sanraku]Interest', interest)
+    end
+    return interest
+end
+
+-- Get Sanraku's "Recommended Fauna"
+xi.znm.getSanrakusFauna = function()
+    local fauna = GetServerVariable('[ZNM][Sanraku]Fauna')
+
+    -- Initialize the server var if it hasn't been already
+    if fauna == nil or fauna == 0 then
+        fauna = math.random(#xi.znm.SANRAKUS_FAUNA)
+        SetServerVariable('[ZNM][Sanraku]Fauna', fauna)
+    end
+    return fauna
+end
+
+--- Does this mob fall under Sanraku's current "Subject of Interest"?
+-- Some families had multiple famIDs (see Sea Monks), and some superfamIDs were too general (see elementals)
+-- In order to only need to use/store a single ID per mob, a bool was added to check what type of ID is stored
+xi.znm.isCurrentInterest = function(superfamID, famID)
+    local family_row = xi.znm.SANRAKUS_INTEREST[xi.znm.getSanrakusInterest()]
+
+    -- Is the ID stored a family ID or a superfamily ID?
+    if family_row.isSuperFamID == 0 then  -- Check mob's family
+        return family_row.familyID == famID
+    else                                 -- Check mob's superfamily
+        return family_row.familyID == superfamID
+    end
+end
+
+--- Is this mob Sanraku's current "Recommended Fauna"?
+xi.znm.isCurrentFauna = function(mobName, zoneID)
+    local fauna_row = xi.znm.SANRAKUS_FAUNA[xi.znm.getSanrakusFauna()]
+    if fauna_row.zone ~= zoneID then
+        return false
+    else
+        if type(fauna_row.name) == "table" then
+            for iter = 1, #fauna_row.name do
+                if fauna_row.name[iter] == mobName then
+                    return true
+                end
+            end
+        else
+            return fauna_row.name == mobName
+        end
+    end
+    return false
+end
+
 -----------------------------------
 -- Soultrapper
 -----------------------------------
 xi.znm.soultrapper = xi.znm.soultrapper or {}
 
+-----------------------------------
+-- onItemCheck
```

Comment:
No need for big header comments with the name of the function. This doesn't add any additional info. The existing block comments are for different sections

Reply:
I'll remove, but I find they provide a nice "reading" break in the code.